fifo.go

```diff
@@ -135,6 +135,7 @@ func openFifo(ctx context.Context, fn string, flag int, perm os.FileMode) (*fifo
 	go func() {
 		if wg != nil {
 			defer wg.Done()
+			runtime.LockOSThread()
```

Comment:
Well. Go runtime will create new thread if all existing threads are in kernel mode. The openAt syscall on fifo won't last too long in practise because kubelet always starts container after created https://github.com/containerd/containerd/pull/11128.

I would like to wait for Go runtime to handle idle threads instead of using LockOSThread here, because it may cause unexpected issue. For example, in process X, sub-thread A invokes exec process Y with `Pdeathsig: SIGKILL`. If the sub-thread A is killed by LockOSThread, the process Y will be killed as well. It's hard to tell who sends signal to process Y in production. So, I am not a fan to use LockOSThread.